FAILED: MultiDcClusterSharding

https://jenkins.akka.io:8498/job/akka-artery-cluster-tests/880/consoleFull

```
[JVM-4] - must join cluster (on node 'fourth', class akka.cluster.sharding.MultiDcClusterShardingMultiJvmNode4) *** FAILED ***
[JVM-4]   false was not true (MultiDcClusterShardingSpec.scala:94)
[JVM-1] [DEBUG] [09/27/2017 12:38:54.716] [New I/O worker #2] [akka.remote.testconductor.ConductorHandler(akka://MultiDcClusterShardingSpec)] message from /10.0.1.235:42889: EnterBarrier(after-2,Some(29999979321 nanoseconds))
[JVM-4] [DEBUG] [09/27/2017 12:38:54.725] [New I/O worker #1] [akka.remote.testconductor.PlayerHandler(akka://MultiDcClusterShardingSpec)] message from a4.moxie/10.0.1.234:4711: BarrierResult(after-2,false)
[JVM-1] [DEBUG] [09/27/2017 12:38:54.726] [New I/O worker #4] [akka.remote.testconductor.PlayerHandler(akka://MultiDcClusterShardingSpec)] message from /10.0.1.234:4711: BarrierResult(fourth-joined,false)
[JVM-3] [DEBUG] [09/27/2017 12:38:54.727] [New I/O worker #1] [akka.remote.testconductor.PlayerHandler(akka://MultiDcClusterShardingSpec)] message from a4.moxie/10.0.1.234:4711: BarrierResult(fourth-joined,false)
[JVM-2] [DEBUG] [09/27/2017 12:38:54.727] [New I/O worker #1] [akka.remote.testconductor.PlayerHandler(akka://MultiDcClusterShardingSpec)] message from a4.moxie/10.0.1.234:4711: BarrierResult(fourth-joined,false)
[JVM-1] [ERROR] [09/27/2017 12:38:54.726] [MultiDcClusterShardingSpec-akka.actor.default-dispatcher-9] [akka://MultiDcClusterShardingSpec/user/controller/barriers] RoleName(fourth) tried to enter 'after-2' while we were waiting for 'fourth-joined' (akka.remote.testconductor.BarrierCoordinator$WrongBarrier: RoleName(fourth) tried to enter 'after-2' while we were waiting for 'fourth-joined')
[JVM-4] - must initialize shards (on node 'fourth', class akka.cluster.sharding.MultiDcClusterShardingMultiJvmNode4) *** FAILED ***
[JVM-4]   java.lang.RuntimeException: barrier failed: after-2
[JVM-4]   at akka.remote.testconductor.ClientFSM$$anonfun$3.applyOrElse(Player.scala:222)
[JVM-4]   at akka.remote.testconductor.ClientFSM$$anonfun$3.applyOrElse(Player.scala:197)
[JVM-4]   at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
[JVM-4]   at akka.actor.FSM$class.processEvent(FSM.scala:665)
[JVM-4]   at akka.remote.testconductor.ClientFSM.akka$actor$LoggingFSM$$super$processEvent(Player.scala:157)
[JVM-4]   at akka.actor.LoggingFSM$class.processEvent(FSM.scala:801)
[JVM-4]   at akka.remote.testconductor.ClientFSM.processEvent(Player.scala:157)
[JVM-4]   at akka.actor.FSM$class.akka$actor$FSM$$processMsg(FSM.scala:659)
[JVM-4]   at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:653)
[JVM-4]   at akka.actor.Actor$class.aroundReceive(Actor.scala:517)
```

Previous case, #23582, was closed 2 days ago, so maybe it wasn't yet fixed?

--- a/akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/MultiDcClusterShardingSpec.scala
+++ b/akka-cluster-sharding/src/multi-jvm/scala/akka/cluster/sharding/MultiDcClusterShardingSpec.scala
@@ -5,13 +5,11 @@
 package akka.cluster.sharding
 
 import scala.concurrent.duration._
-
 import akka.actor.Actor
 import akka.actor.ActorRef
 import akka.actor.Address
 import akka.actor.Props
-import akka.cluster.Cluster
-import akka.cluster.MemberStatus
+import akka.cluster.{ Cluster, MemberStatus, MultiNodeClusterSpec }
 import akka.cluster.sharding.ShardRegion.CurrentRegions
 import akka.cluster.sharding.ShardRegion.GetCurrentRegions
 import akka.remote.testconductor.RoleName
@@ -56,12 +54,16 @@ object MultiDcClusterShardingSpecConfig extends MultiNodeConfig {
 
   commonConfig(ConfigFactory.parseString(s"""
     akka.loglevel = DEBUG # issue #23741
-    akka.cluster.debug.verbose-heartbeat-logging = on
-    akka.cluster.debug.verbose-gossip-logging = on
-    akka.actor.provider = "cluster"
+    akka.cluster {
+      debug.verbose-heartbeat-logging = on
+      debug.verbose-gossip-logging = on
+      auto-down-unreachable-after = 0s
+      sharding {
+        retry-interval = 200ms
+      }
+    }
     akka.remote.log-remote-lifecycle-events = on
-    akka.cluster.auto-down-unreachable-after = 0s
-    """))
+    """).withFallback(MultiNodeClusterSpec.clusterConfig))
 
   nodeConfig(first, second) {
     ConfigFactory.parseString("akka.cluster.multi-data-center.self-data-center = DC1")
@@ -72,19 +74,15 @@ object MultiDcClusterShardingSpecConfig extends MultiNodeConfig {
   }
 }
 
-class MultiDcClusterShardingMultiJvmNode1 extends MultiDcClusterShardingSpec
-class MultiDcClusterShardingMultiJvmNode2 extends MultiDcClusterShardingSpec
-class MultiDcClusterShardingMultiJvmNode3 extends MultiDcClusterShardingSpec
-class MultiDcClusterShardingMultiJvmNode4 extends MultiDcClusterShardingSpec
+class MultiDcClusterShardingSpecMultiJvmNode1 extends MultiDcClusterShardingSpec
+class MultiDcClusterShardingSpecMultiJvmNode2 extends MultiDcClusterShardingSpec
+class MultiDcClusterShardingSpecMultiJvmNode3 extends MultiDcClusterShardingSpec
+class MultiDcClusterShardingSpecMultiJvmNode4 extends MultiDcClusterShardingSpec
 
-abstract class MultiDcClusterShardingSpec extends MultiNodeSpec(MultiDcClusterShardingSpecConfig)
+abstract class MultiDcClusterShardingSpec extends MultiNodeSpec(MultiDcClusterShardingSpecConfig) with MultiNodeClusterSpec
   with STMultiNodeSpec with ImplicitSender {
   import MultiDcClusterShardingSpec._
   import MultiDcClusterShardingSpecConfig._
-
-  override def initialParticipants = roles.size
-
-  val cluster = Cluster(system)
 
   def join(from: RoleName, to: RoleName): Unit = {
     runOn(from) {
@@ -131,8 +129,10 @@ abstract class MultiDcClusterShardingSpec extends MultiNodeSpec(MultiDcClusterSh
       join(fourth, first)
 
       awaitAssert({
-        Cluster(system).state.members.size should ===(4)
-        Cluster(system).state.members.map(_.status) should ===(Set(MemberStatus.Up))
+        withClue(s"Members: ${Cluster(system).state}") {
+          Cluster(system).state.members.size should ===(4)
+          Cluster(system).state.members.map(_.status) should ===(Set(MemberStatus.Up))
+        }
       }, 10.seconds)
 
       runOn(first, second) {

--- a/akka-cluster/src/multi-jvm/scala/akka/cluster/MultiNodeClusterSpec.scala
+++ b/akka-cluster/src/multi-jvm/scala/akka/cluster/MultiNodeClusterSpec.scala
@@ -92,7 +92,7 @@ object MultiNodeClusterSpec {
 
 trait MultiNodeClusterSpec extends Suite with STMultiNodeSpec with WatchedByCoroner with FlightRecordingSupport { self: MultiNodeSpec ⇒
 
-  override def initialParticipants = roles.size
+  final override def initialParticipants = roles.size
 
   private val cachedAddresses = new ConcurrentHashMap[RoleName, Address]
 
@@ -237,14 +237,14 @@ trait MultiNodeClusterSpec extends Suite with STMultiNodeSpec with WatchedByCoro
     def memberInState(member: Address, status: Seq[MemberStatus]): Boolean =
       clusterView.members.exists { m ⇒ (m.address == member) && status.contains(m.status) }
 
-    cluster join joinNode
+    cluster.join(joinNode)
     awaitCond({
       clusterView.refreshCurrentState()
       if (memberInState(joinNode, List(MemberStatus.up)) &&
         memberInState(myself, List(MemberStatus.Joining, MemberStatus.Up)))
         true
       else {
-        cluster join joinNode
+        cluster.join(joinNode)
         false
       }
     }, max, interval)

